File: box/src/main/java/de/qabel/box/storage/IndexNavigation.java
package de.qabel.box.storage;

import de.qabel.box.storage.exceptions.QblStorageException;

import java.util.List;

public interface IndexNavigation extends BoxNavigation {
    List<BoxShare> listShares() throws QblStorageException;

    void insertShare(BoxShare share) throws QblStorageException;

    void deleteShare(BoxShare share) throws QblStorageException;
}

File: box/src/main/java/de/qabel/box/storage/IndexNavigation.kt
package de.qabel.box.storage

import de.qabel.box.storage.exceptions.QblStorageException

interface IndexNavigation : BoxNavigation {
    @Throws(QblStorageException::class)
    fun listShares(): List<BoxShare>

    @Throws(QblStorageException::class)
    fun insertShare(share: BoxShare)

    @Throws(QblStorageException::class)
    fun deleteShare(share: BoxShare)
}
